cassava-megaparsec.cabal

```diff
@@ -1,7 +1,7 @@
+cabal-version:        3.0
```

Comment:
@cptrodolfox are we using a particular feature of cabal available only in 3.x?

Reply:
I agree, increasing `cabal-version` should be motivated.  Since you are dropping support for GHC 8.0, you could naturally raise this to [2.0](https://cabal.readthedocs.io/en/stable/file-format-changelog.html#cabal-version-2-0), as the "native" `Cabal` version for GHC 8.0 is 2.0, see https://www.snoyman.com/base/.
Raising this all the way to 3.0 will make it unnecessarily hard to build this with `stack` on GHC < 8.8, because `stack` LTS snapshots  include just the native version of `Cabal`.

Reply:
@andreasabel Thanks for the suggestion. Totally forgot that the cabal version is tied to the GHC version too will change the version to 2.0.